contracts/governance/extensions/GovernorSequentialProposalId.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Governor} from "../Governor.sol";
+
+abstract contract GovernorSequentialProposalId is Governor {
+    uint256 private _proposalCount;
+    mapping(uint256 proposalHash => uint256 proposalId) private _proposalIds;
+
+    /**
+     * @dev The proposal count may only be set if the current proposal count is 0 in {_setProposalCount}.
+     */
+    error GovernorCanNotSetProposalCount();
```

Comment:
I'd rename it "uninitialized" since it better indicates what's wrong. Also, it goes well with the change I'm proposing to rename {_setProposalCount} to {_initializeProposalCount}. I think that instead of explaining the error literally, we should give a hint to solve it.

```suggestion
    /**
     * @dev The proposal count may only be set if the current proposal count is uninitialized.
     */
    error GovernorUninitializedProposalCount();
```

Reply:
then the error would be `GovernorAlreadyInitializedProposalCount` ?